src/components/inputs/reactQueryBuilder/FieldSelector.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { FullOption, toFlatOptionArray, ValueSelectorProps } from 'react-querybuilder';
+import { Autocomplete, TextField } from '@mui/material';
+
+function FieldSelector(props: ValueSelectorProps) {
```

Comment:
```suggestion
function FieldSelector(props: ReadOnly<ValueSelectorProps>) {
```
There is new control about this in sonar now. But it looks a good idea anyway.

Reply:
yes i saw that too. Thanks i will modify this.